### ISSUE TITLE
Find a way to get rid of duplicate commands.

### ISSUE THREAD

**Issue by 17451k** [17451k/clade#25]
Some commands are in fact just wrappers for other commands. For example, on macOS calling gcc can result in the following command stack:

```
- /usr/bin/gcc ...
- /Library/Developer/CommandLineTools/usr/bin/gcc ...
- /usr/bin/xcrun clang ...
- /Library/Developer/CommandLineTools/usr/bin/clang ...
- /Library/Developer/CommandLineTools/usr/bin/clang -cc1 ...
```

Generally we are interested only in the first command in such stack, but currently there is no way to know that these commands are connected to each other, so corresponding extensions (CC, in case of this example) process all of them as independent commands. The result is several duplicate commands.

**Comment by 17451k** [comment]
Perhaps process id and parent process id may help here.

**Comment by 17451k** [comment]
Implemented in 1022ca4cf4ee93120adbb4ffd3b3b3483c9fea80.

Instead of PID and PPID, custom process ids (and parent ids) were used.
